src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.AddProperty.cs

```diff
@@ -67,7 +94,13 @@ private JsonPropertyInfo AddProperty(Type propertyType, PropertyInfo propertyInf
             return jsonInfo;
         }
 
-        internal static JsonPropertyInfo CreateProperty(Type declaredPropertyType, Type runtimePropertyType, PropertyInfo propertyInfo, Type parentClassType, JsonSerializerOptions options)
+        internal static JsonPropertyInfo CreateProperty(
+            Type declaredPropertyType,
+            Type runtimePropertyType,
+            Type implementedPropertyType,
```

Comment:
How is implementedPropertyType different than runtimePropertyType? runtimePropertyType is intended to be determined at runtime...

Reply:
I've removed the `TImplementedType` generic parameter because we don't need it.

The implementedPropertyType is one of the types that the query type derives from. We select it based on the best fit for our deserialization logic. If the query type is one which the serializer natively supports, the implemented type is the query type itself.
The implemented type is sometimes the same as the runtime property type e.g.

```C#
public class StringListWrapper : List<string> { }
```
Declared type: `typeof(StringListWrapper)`
Implemented type: `typeof(List<string>)`
Runtime type: `typeof(List<string>)`

In some cases the runtime property type is different:

```C#
public class GenericIListWrapper<T> : IList<T>
{
    private readonly List<T> _list = new List<T>();
    
    // IList<T> implementation
    // ...
}
```

Declared type: `typeof(GenericIListWrapper<T>)`
Implemented type: `typeof(IList<T>)`
Runtime type: `typeof(List<T>)`

In this example, this is because we treat IList<T> as List<T>:

Declared type: `typeof(IList<T>)`
Implemented type: `typeof(IList<T>)`
Runtime type: `typeof(List<T>)`

Keeping track of the implemented property type helps us support these derived types e.g here https://github.com/dotnet/corefx/blob/cb54f66c1e0f19d4b97fb19e5326a03d15415776/src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs#L207